crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -45,7 +51,190 @@ impl_downcast!(Stage);
 ///
 /// The checker may report a system more times than the amount of constraints it would actually need
 /// to have unambiguous order with regards to a group of already-constrained systems.
-pub struct ReportExecutionOrderAmbiguities;
+///
+/// By default only a warning with the number of unresolved ambiguities detected will be reported per [`SystemStage`].
+/// This behavior can be changed by explicitly adding this resource using the following constructors:
+/// * [ReportExecutionOrderAmbiguities::off()] - Disables all messages reported by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::minimal()] - Displays only the number of unresolved ambiguities detected by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::verbose()] - Displays a full report of ambiguities detected by the ambiguity checker.
+///
+/// The ambiguity checker will ignore ambiguities within official Bevy crates.
+/// To ignore a custom crate, use [`ReportExecutionOrderAmbiguities::ignore`]
+/// with an list of crate names as an argument.
+///
+/// ## Example
+/// ```ignore
+/// # use bevy_app::App;
+/// # use bevy_ecs::schedule::ReportExecutionOrderAmbiguities;
+/// App::new()
+///    .insert_resource(ReportExecutionOrderAmbiguities::verbose().ignore(&["my_external_crate"]));
+/// ```
+pub struct ReportExecutionOrderAmbiguities {
+    pub level: AmbiguityReportLevel,
+    pub ignore_crates: Vec<String>,
+}
+
+const BEVY_CRATES: [&str; 109] = [
```

Comment:
Having thought about this a bit more, I think this is moving us in the wrong direction for a number of reasons:

1. Moves "bevy engine" crate info "upstream" into `bevy_ecs`, which should be standalone
2. Compiles a big list of strings into bevy_ecs binaries
3. Centralizes what should be decentralized

I personally think we should remove this list in favor of each individual crate calling `ambiguities.ignore("CRATE_NAME_HERE")` _if_ we agree that ambiguities should be ignored for a given crate.

Reply:
Clever. I think this is the best path forward here.

Reply:
Fair points, but since the ignored crates list ate stored in a resource, all crates should have a startup system which will add their crate names to it?

Also, we should add this to all bevy internal crates or only the ones which already have ambiguities reported? 

And lastly, is on the scope of this PR, add this info on all needed crates (I can do it, just wanna organize)

Reply:
> Fair points, but since the ignored crates list are stored in a resource, all crates should have a startup system which will add their crate names to it?

I think it should be a part of Plugin init (prior to the schedule running).

> Also, we should add this to all bevy internal crates or only the ones which already have ambiguities reported?

Yeah I think this is the right call. It encourages us to be more precise about suppressing these. It also cuts down on the amount of work we're doing at startup.

> And lastly, is on the scope of this PR add this info on all needed crates (I can do it, just wanna organize)?

Whatever works best for you!

Reply:
Done. I've added the ignore call on following plugins:
 * bevy_audio
 * bevy_render
 * bevy_sprite
 * bevy_text
 * bevy_ui
 * bevy_transform
 * bevy_winit

The only "trick" part is if users add the `ReportExecutionOrderAmbiguities` after adding `DefaultPlugins` or so, the default ignored crates will be overwritten. I've added this on docs and on example, informing users that if they need to customize the resource, it must be done before the plugins